File: ProjOOPJava/less6/model/DeliveryAction.java
package ProjOOPJava.less6.model;

public class DeliveryAction {
    public void receivingAnOrder(){
        System.out.println("Receiving An Order");
    }
    public void deliveryOrder(){
        System.out.println("Delivery Order");
    }
}
    
class Bike extends DeliveryAction{
    public Bike(){
        System.out.println("Delivery on Bike");
    }
}
class ElectricScooter extends DeliveryAction{
    public ElectricScooter(){
        System.out.println("Delivery on Electric Scooter");
    }
}
class Foots extends DeliveryAction{
    public Foots(){
        System.out.println("Delivery on Foots");
    }
} 
class QuadCopter extends DeliveryAction{
    public QuadCopter(){
        System.out.println("Delivery on Quad Copter");
    }
} 

File: ProjOOPJava/less7/model/DeliveryAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinFormsApp1.model
{
    internal class DeliveryAction
    {
        public String receivingAnOrder()
        {
            return "Получил заказ";
        }
        public String deliveryOrder()
        {
            return "Заказ в доставке";
        }
    }
    abstract class DeliveryGet : DeliveryAction
    {

        public abstract String getTransport();
    }
    class Bike : DeliveryGet
    {
        public override String getTransport()
        {
            return "Доставка выполняется на велосипеде";
        }
    }
    class ElectricScooter : DeliveryGet
    {
        public override String getTransport()
        {
            return "Доставка выполняется на электро самокате";
        }
    }
    class Foots : DeliveryGet
    {
        public override String getTransport()
        {
            return "Доставка выполняется пешком";
        }
    }
    class QuadCopter : DeliveryGet
    {
        public override String getTransport()
        {
            return "Доставка выполняется квадрокоптером";
        }
    }
}
